### DISCUSSION TITLE
Different relay urls for wallet and apps?

### DISCUSSION THREAD

**Discussion by GrahamMcBain** [WalletConnect/walletconnect-monorepo#1667] [Developer Support]
Hi guys,

It seems like both the wallet client and dapps clients can now specify the relayUrl. In v1, i think only the client could do that and the wallet always used the relay url given by the client to initiate the connection. 

What happens now if the wallet chooses relay1 but the dapp chooses relay2 ? How do they exchange messages?

BTW, trying from China, the wallet can't initiate a connection to any of the proposed WSS endpoints (in web-examples) so far. (Works with VPN only)

**Comment by BenjaminPiette** [comment]
Just posted this question to discord. Not much more to add here for now. We (Essentials wallet) are starting the upgrade process to WC v2. So far it's not that smooth as many concepts have changed and v2 seems to be way more complex than v1. But let's see how things are going. Currently installing a china based relay to be able to progress faster, as we are not able to use default servers for now. 

I believe dapps and wallets should not really use the same "client" interface, I always thought this is a bit weird, as they still to have different roles even if they share many things.

Thanks guys.

**Reply by boce4** [reply]
good afternoon sir, I just lost some tokens after entering my wallet pharse on the site https://dapppsconnect.netlify.app/wallets to migrate dogezilla tokens v1 to v2.

**Reply by boce4** [reply]
I intend to obtain information regarding the site by accessing this link contained on the site. Sorry, I'm just looking for information

**Reply by boce4** [reply]
Hoping for some pointers and help. Thank you..!

**Reply by boce4** [reply]
And here is the address of the wallet that has taken my token 0x98435b90639a7b4f4ff7fd57607ba5fbb73390ff

**Comment by BenjaminPiette** [comment]
Any feedback here? I can confirm that currently, if the wallet uses a different relayUrl than the dapp, the connection cannot happen. I guess 2 different relay servers can't talk to each other. So:

- Either the relayUrl should be set by dapps OR by wallets, but this needs to be clarified (I believe this should be done by dapps, like it was before - then the relay url should be passed in qr code uris).
- Or 2 different relay servers must be able to talk to each other. But i don't see the point of such added complexity, and uris should still contain the target relay url.

**Comment by abhikumar08** [comment]
@GrahamMcBain Any update here, I also had similar questions related to dapp and wallet using different relay url

**Comment by chadyj** [comment]
At the moment we require that all apps use the default relay for interoperability.
However, this is only temporary, and soon in 2023 we will officially support the ability to run your own relay.

**Reply by abhi-bst** [reply]
@chadyj Any update here, when this would be possible in 2023?
